### ISSUE TITLE
Date(time) fields don't work with ISO 8601 format when UTC offset is specified

### ISSUE THREAD

**Issue by ltroj** [marcusolsson/obsidian-projects#974]
### What happened?

In my notes i use datetime fields which are formatted according to ISO 8601 including UTC offset like this:

`2024-11-10T16:04+02:00`.

These fields aren't recognized as dates by obisidan-projects for use in calendar views for example. In table view the field is shown as `Text` type.

>Removing the UTC offset like this does seem to work though:

`2024-11-10T16:04`.

### What did you expect to happen?

I Expect datetime fields in the format of `2024-11-10T16:04+02:00`. to be recognized as such.

### How can we reproduce it (as minimally and precisely as possible)?

Create a note with frontmatter:

```
---
date-test: 2024-11-10T16:04+02:00
---
```

and try to use the field date-test for calendar views.

### Plugin version

1.17.4

### Obsidian version

1.7.7

### OS

Windows

**Comment by Acylation** [comment]
Yes thanks for reporting. Will be solved together with issue #940

**Comment by Acylation** [comment]
Note to myself: More work needed towards the actual support of timezone offsets. Now the datetime are majorly taken care of by dayjs, but unfortunately by default it will conduct local conversion when displaying. Possibly be a useful feature, but we'd better make this configurable (trim off the timezone / do the local conversion)

**Comment by Acylation** [comment]
Based on [this post](https://forum.obsidian.md/t/properties-retain-save-iso-8601-time-zone-information/67142) and my experiment, the properties UI just trims off the timezone when displaying zoned datetime values, while after editing thru the picker, the zone info will be erased.

I'm thinking of setting the same behavior as default, and provides more resonable treatments to fully support zones.

**Comment by Acylation** [comment]
Sadly we have to do heavy refactor or write our own time zone parse to support time zone feature. I'm neutral to it, thou neither `day.js` nor native `Date` api persists zone info when they parse the input, `moment.js`, which is used by Obsidian itself, can handle it.

Or we can shift to new `Temporal` api. https://developer.mozilla.org/en-US/blog/javascript-temporal-is-coming/
